bridge/src/commonMain/kotlin/ru/herobrine1st/matrix/bridge/telegram/TelegramWorker.kt

```diff
@@ -40,50 +46,90 @@ class TelegramWorker(
             is ClientEvent.RoomEvent.MessageEvent<*> -> {
                 when (val content = event.content) {
                     is RoomMessageEventContent -> {
-                        val textContent = when (content) {
-                            is RoomMessageEventContent.TextBased.Text -> content.body
-                            else -> throw UnhandledEventException(
-                                message = "This event is not delivered due to lack of support",
-                            )
-                        }
+                        val result = when (content) {
+                            is RoomMessageEventContent.TextBased.Text -> {
+                                val textContent = content.body
+                                val bot = getBot(actorId)
+
+                                withContext(Dispatchers.IO) {
+                                    bot.sendMessage(
+                                        chatId = roomId,
+                                        text = "<${event.sender}>: $textContent",
+                                    )
+                                }
+                            }
 
-                        val bot = getBot(actorId)
-                        val result = withContext(Dispatchers.IO) {
-                            bot.sendMessage(
-                                chatId = roomId,
-                                text = "<${event.sender}>: $textContent",
-                            )
-                        }
+                            is RoomMessageEventContent.FileBased.Image -> {
+                                val bot = getBot(actorId)
+                                val caption = "<${event.sender}>: ${content.body}"
+
+                                downloadMatrixMedia(content.url!!) { photoFile ->
+                                    withContext(Dispatchers.IO) {
+                                        bot.sendPhoto(
+                                            chatId = roomId,
+                                            photo = photoFile,
+                                            caption = caption,
+                                        ).also { it -> it.second?.let<Exception, Unit> { throw it } }
```

Comment:
Forgot to say, this is not needed

```suggestion
                                        )
```